pkg/controller/seed-controller-manager/constraint-controller/controller.go

```diff
@@ -216,15 +226,31 @@ func (r *reconciler) patchFinalizer(ctx context.Context, constraint *kubermaticv
 
 func (r *reconciler) reconcile(ctx context.Context, constraint *kubermaticv1.Constraint, log *zap.SugaredLogger) error {
 
+	clusterList := &kubermaticv1.ClusterList{}
+	if err := r.seedClient.List(ctx, clusterList, &ctrlruntimeclient.ListOptions{LabelSelector: r.workerNameLabelSelector}); err != nil {
+		return fmt.Errorf("failed listing clusters: %w", err)
+	}
+
+	desiredClusterList, unwantedClusterList, err := r.filterClustersForConstraint(ctx, constraint, clusterList)
```

Comment:
nit: I tend to shorten a bit the name `List` is superfluous IMO, I would use `desiredClusters`, `unwantedClusters`